scripts/release/homebrew/docker/formula_generate.py

```diff
@@ -124,6 +125,10 @@ def update_formula() -> str:
     resp.raise_for_status()
     text = resp.text
 
+    # update python version
+    text = re.sub('depends_on "python@.*"', f'depends_on "python@{PYTHON_VERSION}"', text, 1)
+    text = re.sub(r'virtualenv_create\(libexec, "python.*"', f'virtualenv_create(libexec, "python{PYTHON_VERSION}"', text, 1)  # pylint: disable=line-too-long
+
```

Comment:
Should these lines be dropped in the next release? It is really inconvenient to make any change to the formula.

Or, can we just keep the formula as is and make the change when a PR is submitted for https://github.com/Homebrew/homebrew-core/blob/master/Formula/a/azure-cli.rb ?

In my thought, the Azure CLI repo should be self-contained - it should never depend on external resources to build and install.

Reply:
I prefer to keep it here for two reasons.
1. `TestHomebrewFormula` can check whether new formula works.
2. Homebrew users can use the new Python version as soon as possible.

For other changes like the `without_pip: false` in https://github.com/Azure/azure-cli/pull/27186, we can create a PR for homebrew.

Sometimes the Homebrew community update our formula, which cause a conflict, so we modify the external formula. This behavior is introduced in https://github.com/Azure/azure-cli/pull/11964